src/components/common/ProductBox/ProductBox.js

```diff
@@ -22,25 +22,24 @@ const ProductBox = ({
   compare,
   addToFavourite,
   heart,
-  addCompare,
 }) => {
   const compareHandler = event => {
     event.preventDefault();
     const maxProductsToCompare = 4;
-    const isProductAddedToCompare =
-      compare.products &&
-      compare.products.reduce(
-        (accumulator, product) => accumulator || product.id === id,
-        false
-      );
-
     if (isProductAddedToCompare !== true) {
       count < maxProductsToCompare
         ? setCompare({ id, image })
         : alert(`You can compare maximum of ${maxProductsToCompare} products!`);
     }
   };
 
+  const isProductAddedToCompare =
```

Comment:
Przerzuć tą deklarcję ponad metodę `compareHandler` bo dziwnie to wygląda, że w metodzie `comparehandler` odwołujemy się do zmiennej która jest deklarowana poniżej metody.